### ISSUE TITLE
isoquant stuck at the read processing for hours

### ISSUE THREAD

**Issue by koryant** [ablab/IsoQuant#69]
Hi there, 

Thanks for creating this awesome tool! When I try to run isoquant on a nanpore long read sequencing data from Mouse. The program is stuck at read processing step for hours. Please advise on how I should trouble shoot. Thank you very much! Look forward to hearing from you!

The command I used is as followed:
```
isoquant.py \
--reference GRCm38.primary_assembly.genome.fa \
--genedb gencode.vM21.primary_assembly.annotation.gtf.db \
--bam_list bam.list \
--data_type nanopore \
--labels S1 S2 S3 S4 S5 \
--read_group file:read_info.csv.gz:0:1:, \
--transcript_quantification all \
--gene_quantification all \
-o isoQuant_output/
```

The first ran is stuck at chromosome 6 and the second ran is stuck at chromosome 14. I pasted the last lines of the output from the second ran below. 

```
2023-02-17 23:29:15,662 - INFO - Collecting read alignments
2023-02-17 23:30:54,635 - INFO - Processing chromosome chr1
2023-02-17 23:31:39,757 - INFO - Processing chromosome chr2
2023-02-17 23:32:30,061 - INFO - Processing chromosome chrX
2023-02-17 23:33:17,654 - INFO - Processing chromosome chr3
2023-02-17 23:34:03,572 - INFO - Processing chromosome chr4
2023-02-17 23:34:49,765 - INFO - Processing chromosome chr5
2023-02-17 23:35:35,242 - INFO - Processing chromosome chr6
2023-02-17 23:37:13,933 - INFO - Processing chromosome chr10
2023-02-17 23:38:04,456 - INFO - Processing chromosome chr8
2023-02-17 23:38:53,973 - INFO - Processing chromosome chr14
```

```
2023-02-17 23:41:59,376 - DEBUG - === Processing read d4bd372f-8a36-4eaf-b67a-904de3bcf606 ===
2023-02-17 23:41:59,376 - DEBUG - PolyA found at position 8265502
2023-02-17 23:41:59,376 - DEBUG - No polyT found
2023-02-17 23:41:59,376 - DEBUG - No polyA found
2023-02-17 23:41:59,376 - DEBUG - No polyT found
2023-02-17 23:41:59,377 - DEBUG - === Finished read d4bd372f-8a36-4eaf-b67a-904de3bcf606 ===
2023-02-17 23:41:59,377 - DEBUG - === Processing read b91caf0e-f0d5-4649-8b70-78ffba3b75d5 ===
2023-02-17 23:41:59,377 - DEBUG - PolyA found at position 8265502
2023-02-17 23:41:59,377 - DEBUG - No polyT found
2023-02-17 23:41:59,377 - DEBUG - No polyA found
2023-02-17 23:41:59,377 - DEBUG - No polyT found
2023-02-17 23:41:59,377 - DEBUG - === Finished read b91caf0e-f0d5-4649-8b70-78ffba3b75d5 ===
2023-02-17 23:41:59,378 - DEBUG - Processing region (8266496, 8299776) with 1 reads
2023-02-17 23:42:00,513 - DEBUG - === Processing read 4d49026b-3f9c-4feb-af6f-3ab8af124d5d ===
2023-02-17 23:42:00,513 - DEBUG - No polyA found
2023-02-17 23:42:00,536 - DEBUG - PolyT found at position 8297293
2023-02-17 23:42:00,536 - DEBUG - No polyA found
2023-02-17 23:42:00,602 - DEBUG - No polyT found
2023-02-17 23:42:00,703 - DEBUG - + Validating polyA/T sites
```

**Comment by andrewprzh** [comment]
Dear @koryant 

Thank you for the positive feedback!

For how long is it stuck? Just in case, could you send me the log file?

For now I can suggest resuming the run with --resume option and check whether it gets stuck in the same place. If it does, then it sounds like a but. If it gets stuck in different places, could a disk IO issue.

Best
Andrey

**Comment by koryant** [comment]
Dear Andrey, 

Thanks so much for the speedy response! The second time I ran was stuck for over 10 hours, so I stopped it.  I tried three times. The firs two times seemed to be stuck at the same place (the chromosome 6 of the mouse genome) and the third time was stuck at chromosome 14. So it may be a disk IO issue. I will try it again. I also sent you the log file via email. Thanks again for the help!
